javascripts/discourse/components/modal/insert-video.js

```diff
@@ -0,0 +1,104 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { empty } from "@ember/object/computed";
+import { isVideo } from "discourse/lib/uploads";
+import { afterRender } from "discourse-common/utils/decorators";
+import I18n from "I18n";
+
+export default class InsertVideoModal extends Component {
+  @tracked sources;
+  @tracked tracks;
+  @tracked poster;
+  @empty("sources") insertDisabled;
+
+  _keyDown(e) {
+    if (e.keyCode === 13) {
+      e.preventDefault();
+      e.stopPropagation();
+      return false;
+    }
+  }
+  @afterRender
+  _preventSubmit() {
+    // prevent submitting on enter while adding items to lists using Enter
+    document
+      .querySelector(".video-sources")
+      .addEventListener("keydown", (e) => this._keyDown(e));
+
+    document
+      .querySelector(".video-subtitles")
+      .addEventListener("keydown", (e) => this._keyDown(e));
+  }
```

Comment:
Does this work in a glimmer component? 🤔 even if it does, imo it'd be better to use `{{did-insert}}` modifier to run this.